blueflood-core/src/test/java/com/rackspacecloud/blueflood/utils/ModuleLoaderTest.java

```diff
@@ -0,0 +1,214 @@
+package com.rackspacecloud.blueflood.utils;
+
+import com.rackspacecloud.blueflood.io.DiscoveryIO;
+import com.rackspacecloud.blueflood.io.DummyDiscoveryIO;
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.CoreConfig;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+
+
+public class ModuleLoaderTest {
+
+    @Before
+    public void setUp() {
+        Configuration.getInstance().clearProperty(CoreConfig.DISCOVERY_MODULES);
+        ModuleLoader.clearCache();
+    }
+
+    @After
+    public void tearDown() {
+        Configuration.getInstance().clearProperty(CoreConfig.DISCOVERY_MODULES);
+        ModuleLoader.clearCache();
+    }
+
+    @Test
+    public void singleClassYieldsThatClass() {
+
+        // given
+        Configuration.getInstance().setProperty(
+                CoreConfig.DISCOVERY_MODULES,
+                "com.rackspacecloud.blueflood.utils.DummyDiscoveryIO3");
+
+        // when
+        Object loadedModule = ModuleLoader.getInstance(DiscoveryIO.class, CoreConfig.DISCOVERY_MODULES);
+
+        // then
+        Assert.assertNotNull(loadedModule);
+        Assert.assertEquals(DummyDiscoveryIO3.class, loadedModule.getClass());
+    }
+
+    @Test
+    public void singleClassInDifferentPackageYieldsThatClass() {
```

Comment:
I am thinking of a test case where it would throw "IllegalAccessException". I believe for that we have to pick a class from a different package which is at package-protected access level or may be a private class to trigger that exception.